src/mods/timer.cpp

```diff
@@ -13,7 +14,119 @@ static u32 s_prev_retrace_count;
 static s32 s_rta_timer;
 static s32 s_pause_timer;
 
-void init() { s_retrace_count = mkb::VIGetRetraceCount(); }
+static u32 s_framesave;
+static patch::Tramp<decltype(&mkb::did_ball_enter_goal)> s_goal_tramp;
+
+bool line_intersects(Vec* lineStart, Vec* lineEnd, mkb::Rect* rect) {
```

Comment:
These can probably be const references instead of pointers. Oh and what's the reason for not calling the calling the game's function btw?

Reply:
The calculation to get the subtick is done inside of this function, I didn't want to just hook this function though because I assume it is also used for wormhole intersection tests

Reply:
Gotcha.

Also it would be better to make `s_framesave` a function output (non-const reference argument in this case) than a static variable.